sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -325,17 +325,24 @@ public static ReadRows readRows() {
    *
    * @param <T> Type of the data to be read.
    */
-  public static <T> ReadWithPartitions<T> readWithPartitions() {
-    return new AutoValue_JdbcIO_ReadWithPartitions.Builder<T>()
+  public static <T, PartitionColumnT> ReadWithPartitions<T, PartitionColumnT> readWithPartitions(
```

Comment:
you're right. Indeed, I wanted to avoid breaking as much as possible - so this change does not break the existing integration tests (see JdbcIOIT), while adding support for any type partitioning - and the transform is fairly new, so I think we may be able to pull this off : ) - WDYT?

Reply:
I think this is acceptable and the best possible option 👍   just wanted to raise it ...

Reply:
thanks!